packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts

```diff
@@ -275,12 +279,30 @@ export class AuthService {
     return this.hashToken(token) === hashedToken;
   }
 
+  public prepareWeb3PreSignUpPayload(
+    data: Web3PreSignUpDto,
+  ): Web3PreSignUpPayloadDto {
+    const signer = this.web3Service.getSigner(data.chainId);
+
+    return {
+      from: data.address,
+      to: signer.address,
+      contents: WEB3_SIGNUP_MESSAGE,
+    };
+  }
+
+  public async web3PreSignup(
+    data: Web3PreSignUpDto,
+  ): Promise<Web3PreSignUpPayloadDto> {
+    return this.prepareWeb3PreSignUpPayload(data);
+  }
+
   public async web3Signup(data: Web3SignUpDto): Promise<AuthDto> {
-    const verified = await verifySignature(
-      WEB3_SIGNUP_MESSAGE,
-      data.signature,
-      [data.address],
-    );
+    const preSignUpData = this.prepareWeb3PreSignUpPayload(data);
+
+    const verified = await verifySignature(preSignUpData, data.signature, [
```

Comment:
I am getting Invalid signature when I test it. I am not sure why, but I think you might need to convert the preSignUpData to string

Reply:
Yes, you are right